bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/discovery/WebthingDiscoveryService.java

```diff
@@ -0,0 +1,273 @@
+            // WebThings if a ingle WebTHing has not been found.
+            // Furthermore, first it will be tried to connect the endppint using https. If this fails, as fallback
+            // plain http is used.
+
+            // check single WebThing path via https (e.g. https://192.168.0.23:8433/)
+            var optionalDiscoveryResult = discoverWebThing(toURI(host, port, path, true));
+            if (optionalDiscoveryResult.isPresent()) {
+                discoveryResults.add(optionalDiscoveryResult.get());
+            } else {
+                // check single WebThing path via plain http (e.g. http://192.168.0.23:8433/)
+                optionalDiscoveryResult = discoverWebThing(toURI(host, port, path, false));
+                if (optionalDiscoveryResult.isPresent()) {
+                    discoveryResults.add(optionalDiscoveryResult.get());
+                } else {
+                    // check multiple WebThing path via https (e.g. https://192.168.0.23:8433/0,
+                    // https://192.168.0.23:8433/1,...)
+                    outer: for (int i = 0; i < 50; i++) { // search 50 entries at maximum
+                        optionalDiscoveryResult = discoverWebThing(toURI(host, port, path + i, true));
+                        if (optionalDiscoveryResult.isPresent()) {
+                            discoveryResults.add(optionalDiscoveryResult.get());
+                        } else if (i == 0) {
+                            // check multiple WebThing path via plain http (e.g. http://192.168.0.23:8433/0,
+                            // http://192.168.0.23:8433/1,...)
+                            for (int j = 0; j < 50; j++) { // search 50 entries at maximum
+                                optionalDiscoveryResult = discoverWebThing(toURI(host, port, path + j, false));
+                                if (optionalDiscoveryResult.isPresent()) {
+                                    discoveryResults.add(optionalDiscoveryResult.get());
+                                } else {
+                                    break outer;
+                                }
+                            }
+                        } else {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        return discoveryResults;
+    }
+
+    private Optional<DiscoveryResult> discoverWebThing(URI uri) {
+        try {
+            var description = descriptionLoader.loadWebthingDescription(uri, Duration.ofSeconds(5));
+
+            var id = (uri.getHost() + uri.getPort() + uri.getPath()).replaceAll("\\W", "");
+            var thingUID = new ThingUID(THING_TYPE_UID, id);
+            return Optional.of(DiscoveryResultBuilder.create(thingUID).withThingType(THING_TYPE_UID)
+                    .withProperty("webThingURI", uri).withLabel(description.title).build());
```

Comment:
Can you add the representation property?

Reply:
done